mem.go

```diff
@@ -262,7 +258,7 @@ func ValidUTF8(m RO) bool {
 
 // NewReader returns a new Reader that reads from m.
 func NewReader(m RO) *Reader {
-	return &Reader{sr: strings.NewReader(m.str())}
+	return &Reader{br: bytes.NewReader(m.bytes())}
```

Comment:
What is the reason for switching from `strings.Reader` to `bytes.Reader`?

Reply:
`strings.Reader.WriteTo` have one extra alloc compared to `bytes.Reader.WriteTo`

Reply:
`strings.Reader.WriteTo` will try to use `io.WriteString`, and fallback to `Write([]byte(str))` so it will alloc if writer doesn't implement `io.StringWriter`

Reply:
old todo comment: `// TODO: add Reader.WriteTo, but don't use strings.Reader.WriteTo because it uses io.WriteString, leaking our unsafe string`